src/Subtitles/TextFile.cpp

```diff
@@ -588,36 +588,14 @@ BOOL CTextFile::ReadString(CStringW& str)
                     m_wbuffer[nCharsRead] = m_buffer[m_posInBuffer] & 0x7f;
                 } else if (Utf8::isFirstOfMultibyte(m_buffer[m_posInBuffer])) {
                     int nContinuationBytes = Utf8::continuationBytes(m_buffer[m_posInBuffer]);
-                    ASSERT(bValid);
-
-                    if (m_posInBuffer + nContinuationBytes >= m_nInBuffer) {

```

Comment:
Shouldn't this check still be needed, in case there is data for a partial character at the end of the file?

Reply:
Yes, I need to add that check, and maybe one other to validate the UTF8 bytes.